content/cucumber/step-definitions.md

```diff
@@ -253,3 +253,60 @@ Given("I have {int} {color} balls", function (int1, color) {
 });
 ```
 {{% /block %}}
+
+# How do I group my step definitions?
+
+Avoid writing similar step definitions which can lead to clutter. While documenting your steps helps, making use of **helper methods** to abstract them can do wonders.
+
+For eg, if you have the following steps :
+
+```
+    Given I go to the home page
+    Given I check the about page of the website 
+    Given I get the contact details
+```
+
+and all of these steps visit the **Home**, **About** and **Contact** pages, you might be writing _redundant steps_. While the underlying code for these steps could be different, their **behaviour** is essentially the same, ie.
+
+1. Open the Home page.
+2. Open the About page.
+3. Open the Contact page.
+
+As such, you can use abstract helper methods to reduce all these into one step :
+
+    Given I go to the {} page
+
+with the step def :
+
+```java
+@Given("I want to open the {string} page")
+public void i_want_to_open_page(String name) {
+  pageFactory.openPage(name);
+}
+```
+
+```javascript
+ //TODO
+```
+
+```ruby
+ //TODO
+```
+
+
+```kotlin
+ //TODO
+```
+
+And have your step definitions act as a gateway to the actual code (in this case, a factory method to decide which page to open).
+
+This helps in a number of ways,
+
+1. Less and more easily maintainable steps.
+2. Increased Scalablility: Adding tests for a new functionality with the same underlying _behaviour_ is easier.
+3. Concise and easy to understand tests.
+
+You can use the same method to write steps for validating a webpage, clicking a button, etc each with their own helper methods.
+
+We recommend taking a look at the [Factory Design Pattern] (https://refactoring.guru/design-patterns/factory-method) as well.
```

Comment:
I'd prefer "suggest" over "recommend", as I have no strong feelings on this.

Reply:
Yes that makes sense. Updated.